src/util/net.js

```diff
@@ -0,0 +1,20 @@
+/* @flow */
+let http = require('http');
+let https = require('https');
+
+/**
+ * Fetch a file from a given url
+ * Returns a promise resolving to a tuple of [body, statusCode]
+ * Will throw an error on connection errors
+ */
+export function httpFetchFile(url: string): Promise<[string, number]> {
```

Comment:
This code would be easier to write (and easier to test) if you use the [request](https://github.com/request/request) module. I don't mind adding it as a dependency.